headers/data/overlay00.h

```diff
@@ -0,0 +1,6 @@
+#ifndef HEADERS_DATA_OVERLAY00_H_
```

Comment:
I think we should put these in a new header file, `headers/data/literals.h` to match the structure of the YAML files. I'm pretty sure the test scripts should "just work" (or at least only require minor changes) if we do this.

Reply:
Not sure if I like that approach. I think it has multiple downsides:

- It's not possible to know what binary each symbol belongs to just by looking at the header file
- It's very hard to know what the sort order for new symbols should be just by looking at the header file
- Symbols in the header file would be sorted by their addresses, even though they belong to different binaries. You could get completely unrelated symbols next to each other, multple related symbols split into separate groups, two unrelated symbols with the same address, etc.
- This creates yet another .sym file to import for the runtime debugging setup. It also means that you're forced to choose between importing all literals or none of them, regardless of which binary they belong to. Multiple literals with the same address would also be a problem in this scenario.

I think separating them by binary on the header files works better.

Reply:
> * It's not possible to know what binary each symbol belongs to just by looking at the header file
> 
> * It's very hard to know what the sort order for new symbols should be just by looking at the header file

I feel like this could be solved with comments. We could add comments in `literals.h` labeling each binary.

> * Symbols in the header file would be sorted by their addresses, even though they belong to different binaries. You could get completely unrelated symbols next to each other, multple related symbols split into separate groups, two unrelated symbols with the same address, etc.

This isn't how `symbol_check.py` works. When it parses the YAML files it preserves the order of blocks. So all the arm9 symbols will come first in address order, then the overlay0 symbols, etc.

>     * This creates yet another .sym file to import for the runtime debugging setup. It also means that you're forced to choose between importing all literals or none of them, regardless of which binary they belong to. Multiple literals with the same address would also be a problem in this scenario.

Not really. This is entirely controlled by the YAML files. The header files have nothing to do with it.

> I think separating them by binary on the header files works better.

I would be okay with this approach if all the literals were put in their own section at the end of the header file. But that would require larger changes to `symbol_check.py` that I'm not really keen on making.